openid-4-verifiable-presentations-1_0.md

```diff
@@ -1900,88 +1900,137 @@ these constraints. The Wallet is not controlled by the Verifier and the Verifier
 
-Requests from the Wallet to the Verifier SHOULD be sent with the minimal amount of information possible, and in particular, without any HTTP headers identifying the software used for the request (e.g., HTTP libraries or their versions). The Wallet MUST NOT send PII or any other data that could be used for fingerprinting to the Request URI in order to prevent End-User tracking. 
+Wallets operating within a trust framework SHOULD validate that the Request URI is properly associated with the Client Identifier and authorized for the request.
 
-## Authorization Error Response with the `wallet_unavailable` error code
+Untrusted or unrecognized Request URI endpoints SHOULD be rejected or require End-User confirmation before proceeding.
 
-In the event that another component is invoked instead of the Wallet, the End-User MUST be informed and give consent before the invoked component returns the `wallet_unavailable` Authorization Error Response to the Verifier.
+### Authorization Requests with Request URI {#authorization_requests_with_request_uri}
 
-## Digital Credential API Error Responses {#privacy-dc-api-error}
+If the Wallet is acting within a trust framework that allows the Wallet to determine whether a Request URI belongs to a certain Client Identifier, the Wallet is RECOMMENDED to validate the Verifier's authenticity and authorization given by the Client Identifier and that the Request URI corresponds to this Verifier. If the link cannot be established in those cases, the Wallet MUST refuse the request.
 
-Returning any OpenID4VP protocol error, regardless of content, can reveal additional information about the user’s underlying Credentials or Wallet in a way that is unique to the Digital Credentials API since reaching the Wallet can be dependent on a Wallet's ability to satisfy the request. For example, platform implementations could only allow Wallets to be selected that satisfy the request. In this case, returning an OpenID4VP protocol error responses can only be returned by a selected Wallet and would therefore reveal that the user is in possession of Credentials that satisfy the request. This is in contrast to other engagement methods, where the Wallet receives the request before learning if it can be fulfilled. What is revealed by a Wallet in those cases depends on how each individual Wallet processes a request.
+If no End-User interaction is required before sending the request, it is easy to request on a large scale and in an automated fashion the Wallet capabilities from all visitors of a website. Even without PII this can reveal some information about End-Users, like their nationality (e.g., a Wallet with special capabilities only used in one EU member state).
+
+Mandatory End-User interaction before sending the request, like clicking a button, unlocking the Wallet or even just showing a screen of the app, can make this less attractive/likely to being exploited.
```

Comment:
```suggestion
Mandatory End-User interaction before sending the request, like clicking a button or unlocking the Wallet, can make this less attractive/likely to being exploited.
```

Just showing the screen of the app is not really an End-User interaction from my POV, instead an End-User interaction is something that has to solicit input from the user in some fashion.